api/src/main/scala/hmda/api/protocol/admin/WriteInstitutionProtocol.scala

```diff
@@ -0,0 +1,140 @@
+package hmda.api.protocol.admin
+
+import hmda.api.protocol.processing.InstitutionProtocol
+import hmda.model.institution.Agency._
+import hmda.model.institution.DepositoryType.{ Depository, NonDepository }
+import hmda.model.institution.ExternalIdType._
+import hmda.model.institution.InstitutionType._
+import hmda.model.institution._
+import spray.json._
+
+trait WriteInstitutionProtocol extends InstitutionProtocol {
+
+  implicit object AgencyJsonFormat extends RootJsonFormat[Agency] {
+
+    override def write(obj: Agency): JsValue = obj match {
+      case CFPB => JsString(CFPB.name)
+      case FDIC => JsString(FDIC.name)
+      case FRS => JsString(FRS.name)
+      case HUD => JsString(HUD.name)
+      case NCUA => JsString(NCUA.name)
+      case OCC => JsString(OCC.name)
+    }
+
+    override def read(json: JsValue): Agency = json match {
+      case JsString(CFPB.name) => CFPB
+      case JsString(FDIC.name) => FDIC
+      case JsString(FRS.name) => FRS
+      case JsString(HUD.name) => HUD
+      case JsString(NCUA.name) => NCUA
+      case JsString(OCC.name) => OCC
+      case _ => throw DeserializationException("Unable to deserialize")
+    }
+
+  }
+
+  implicit object InstitutionTypeJsonFormat extends RootJsonFormat[InstitutionType] {
```

Comment:
Huh.  I made these changes locally, and I _thought_ all the tests passed.  Anyway, not worth holding up the PR over.  I'll create another issue to remind me to look back into it.